demo-apps/cf-simplefile-server/src/main/java/org/eclipse/californium/examples/SimpleFileServer.java

```diff
@@ -0,0 +1,202 @@
+			try {
+				super.handleRequest(exchange);
+			} catch (Exception e) {
+				LOG.log(Level.SEVERE, "Exception while handling a request on the " + getName() + " resource", e);
+				exchange.sendResponse(new Response(CoAP.ResponseCode.INTERNAL_SERVER_ERROR));
+			}
+		}
+
+		@Override
+		public void handleGET(final CoapExchange exchange) {
+			Request request = exchange.advanced().getRequest();
+			LOG.log(Level.INFO, "Get received : {0}", request);
+
+			if (null == dataRoot) {
+				exchange.respond(CoAP.ResponseCode.NOT_FOUND);
+				return;
+			}
+
+			int accept = request.getOptions().getAccept();
+			if (MediaTypeRegistry.UNDEFINED == accept) {
+				accept = MediaTypeRegistry.APPLICATION_OCTET_STREAM;
+			} else if (MediaTypeRegistry.APPLICATION_OCTET_STREAM != accept) {
+				exchange.respond(CoAP.ResponseCode.UNSUPPORTED_CONTENT_FORMAT);
+				return;
+			}
+
+			String myURI = getURI() + "/";
+			String path = "/" + request.getOptions().getUriPathString();
+			if (!path.startsWith(myURI)) {
+				LOG.log(Level.INFO, "Request {0} doesn't match {1}!", new Object[] { path, myURI });
+				exchange.respond(CoAP.ResponseCode.INTERNAL_SERVER_ERROR);
+				return;
+			}
+			path = path.substring(myURI.length());
+			if (request.getOptions().hasBlock2()) {
+				LOG.log(Level.INFO, "Send file {0} {1}", new Object[] { path, request.getOptions().getBlock2() });
+			} else {
+				LOG.log(Level.INFO, "Send file {}", path);
+			}
+			File file = new File(dataRoot, path);
+			if (!file.exists() || !file.isFile()) {
+				LOG.log(Level.WARNING, "File {0} doesn't exist!", file.getAbsolutePath());
+				exchange.respond(CoAP.ResponseCode.NOT_FOUND);
+				return;
+			}
+			if (!file.canRead()) {
+				LOG.log(Level.WARNING, "File {0} is not readable!", file.getAbsolutePath());
+				exchange.respond(CoAP.ResponseCode.UNAUTHORIZED);
+				return;
+			}
+			long length = file.length();
+			if (length > MAX_FILE_SIZE) {
+				LOG.log(Level.WARNING, "File {0} is too large {1} (max.: {2})!",
+						new Object[] { file.getAbsolutePath(), length, MAX_FILE_SIZE });
+				exchange.respond(CoAP.ResponseCode.INTERNAL_SERVER_ERROR);
+				return;
+			}
+			try (InputStream in = new FileInputStream(file)) {
+				byte[] content = new byte[(int) length];
+				int r = in.read(content);
+				if (length == r) {
+					Response response = new Response(CoAP.ResponseCode.CONTENT);
+					response.setPayload(content);
+					response.getOptions().setSize2((int) length);
+					response.getOptions().setContentFormat(accept);
+					exchange.respond(response);
+				} else {
+					LOG.log(Level.WARNING, "File {0} could not be read in!", file.getAbsolutePath());
+					exchange.respond(CoAP.ResponseCode.INTERNAL_SERVER_ERROR);
+				}
+			} catch (IOException ex) {
+				LOG.log(Level.WARNING, "File {0}: {1}", new Object[] { file.getAbsolutePath(), ex });
+				exchange.respond(CoAP.ResponseCode.INTERNAL_SERVER_ERROR);
+			} catch (OutOfMemoryError error) {
```

Comment:
IOException of OutOfMemoryError?
It's easy, I just want to have some logging and defined responses in the "exceptional" cases even if they never occur. But no problem we can remove one or both